src/api/db/models/Project.ts

```diff
@@ -334,6 +367,49 @@ export class ProjectModel {
     }
   }
 
+  static async removeCameraFromViews(
+    input: { cameraId: string },
+    context: Pick<Context, 'user'>,
+  ): Promise<HydratedDocument<ProjectSchema>> {
+    try {
+      return await retry(
+        async () => {
+          // find project
+          let project = await Project.findOne({ _id: context.user['curr_project'] });
+          if (!project) throw new NotFoundError('Project not found');
+
+          // get all deployment ids for the camera
+          const projectDeps =
+            project.cameraConfigs
+              .find((cc) => cc._id === input.cameraId)
+              ?.deployments.map((d) => d._id.toString()) ?? [];
+
+          console.log('deployments to be removed from views: ', projectDeps);
+
+          project.views.forEach((v, index) => {
+            // if view filters has the camera id or any of the deployment ids
+            if (
+              v.filters.cameras?.includes(input.cameraId) ||
+              projectDeps.some((d) => v.filters.deployments?.includes(d))
+            ) {
+              v.filters.cameras = v.filters.cameras?.filter((c) => c === input.cameraId);
```

Comment:
I think we want a `filter((c) => c !== input.cameraId)` here, right?

Reply:
You're right, didn't actually catch this because currently there's no way on the frontend to add a camera to the filters.